src/utils/io/progress_log.rs

```diff
@@ -115,18 +138,57 @@ impl Progress {
         self.emit_log_if_required();
     }
 
+    // Example output:
+    //
+    // Bytes, with total: 12.4 MiB / 1.2 GiB, 1%, 1.5 MiB/s, elapsed time: 8m 12s
+    // Bytes, without total: 12.4 MiB, 1.5 MiB/s, elapsed time: 8m 12s
+    // Items, with total: 12 / 1200, 1%, 1.5 items/s, elapsed time: 8m 12s
+    // Items, without total: 12, 1.5 items/s, elapsed time: 8m 12s
+    fn msg(&self, now: Instant) -> String {
+        let elapsed_secs = (now - self.start).as_secs_f64();
+        let elapsed_duration = format_duration(Duration::from_secs(elapsed_secs as u64));
+
+        let mut output = String::new();
```

Comment:
It might be more readable to collect all the portions of the message and then, at the end, have a `format!("{message} {item_type} {total} {speed}, elapsed time {elapsed}"). Or something like this.

Reply:
Done.